core/src/main/scala/akka/kafka/javadsl/Producer.scala

```diff
@@ -31,6 +33,20 @@ object Producer {
       .asJava
 
   /**
+   * The `plainSink` can be used for publishing records to Kafka topics.
+   * The `record` contains a topic name to which the record is being sent, an optional
+   * partition number, and an optional key and value.
+   */
+  def plainSink[K, V](
+    producerProvider: () => KafkaProducer[K, V],
```

Comment:
The only problem is a impossibility to create different `KafkaProducer` for multiple stream runs that created with ProducerProvider. Yes producer is thread safe, but may be it can be useful. Do we care about this scenario?